core/includes/install.core.inc

```diff
@@ -1453,6 +1451,313 @@ function install_select_language_form($form, &$form_state, $files) {
+      ];
+    }
+    else {
+      $requirements['translations directory writable'] = [
+        'title'       => t('Translations directory'),
+        'value'       => t('The translations directory is writable.'),
+      ];
+    }
+  }
+
+  // If the translations server can not be contacted, throw an error.
+  if (!$online) {
+    $requirements['online'] = [
+      'title'       => t('Internet'),
+      'value'       => t('The translation server is offline.'),
+      'severity'    => REQUIREMENT_ERROR,
+      'description' => t('The installer requires to contact the translation server to download a translation file. Check your internet connection and verify that your website can reach the translation server at <a href="!server_url">@server_url</a>.', ['!server_url' => $server_url, '@server_url' => $server_url]),
+    ];
+  }
+  else {
+    $requirements['online'] = [
+      'title'       => t('Internet'),
+      'value'       => t('The translation server is online.'),
+    ];
+    // If translation file is not found at the translation server, throw an
+    // error.
+    if (!$translation_available) {
+      $requirements['translation available'] = [
+        'title'       => t('Translation'),
+        'value'       => t('The %language translation is not available.', ['%language' => $language]),
+        'severity'    => REQUIREMENT_ERROR,
+        'description' => t('The %language translation file is not available at the translation server. <a href="!url">Choose a different language</a> or select English and translate your website later.', ['%language' => $language, '!url' => $_SERVER['SCRIPT_NAME']]),
+      ];
+    }
+    else {
+      $requirements['translation available'] = [
+        'title'       => t('Translation'),
+        'value'       => t('The %language translation is available.', ['%language' => $language]),
+      ];
+    }
+  }
+
+  if ($translations_directory_exists && $readable && $writable && $translation_available) {
+    $translation_downloaded = install_retrieve_file($translation_url, $translations_directory);
+
+    if (!$translation_downloaded) {
+      $requirements['translation downloaded'] = [
+        'title'       => t('Translation'),
+        'value'       => t('The %language translation could not be downloaded.', ['%language' => $language]),
+        'severity'    => REQUIREMENT_ERROR,
+        'description' => t('The %language translation file could not be downloaded. <a href="!url">Choose a different language</a> or select English and translate your website later.', ['%language' => $language, '!url' => $_SERVER['SCRIPT_NAME']]),
```

Comment:
I wonder, if we can't just move on (without translation file) here.
There are several reasons for a download failure, and I'm not sure if it's really necessary to prevent setting language and locale modules up with a language setting, for which we don't have "official" translations, yet. Just as an example.

Of course, the interface is then still English, so we might want to inform people that we didn't get a .po file. But the rest of the setup doesn't break by a missing po.
Not sure...

Reply:
I'm thinking we should just default to an array of the 27 languages available right now, and update that list with every release. So users dont have to guess which languages are available.

Eventually we could code on the localization server to also serve a plain list of available languages, and query that during install to build the list of available languages.

Reply:
So I've limited the languages on the select list to only those available on our localize server. I think this is a cleaner approach.

But looking again, I think we should leave this check here now, because it indicates to the user that there *is* a language file but it cant be downloaded (server issues, no internet etc) so they can try to find it manually.

Reply:
> So I've limited the languages on the select list to only those available on our localize server. 

Sure, let's try that. See also standard_language_list() (which might need an update).

> I think we should leave this check here now, because it indicates to the user that there is a language file but it cant be downloaded (server issues, no internet etc)

Seems reasonable to me.